src/integrationTest/java/uk/gov/hmcts/reform/blobrouter/services/storage/ContainerCleanerIntTest.java

```diff
@@ -0,0 +1,187 @@
+            .containsExactlyInAnyOrder(dispatchedFileNames);
+
+        createEnvelopes(DISPATCHED, dispatchedFileNames);
+
+        // when
+        containerCleaner.process(CONTAINER_NAME);
+
+        // then
+        assertThat(containerClient.listBlobs()).isEmpty();
+        assertFilesIsDeleteState(true, dispatchedFileNames);
+    }
+
+    @Test
+    public void should_delete_only_dispatched_blobs() {
+        // given
+        String[] dispatchedFileNames = new String[]{TEST_1, TEST_3};
+        String[] rejectedFileNames = new String[]{TEST_2, TEST_4};
+
+        uploadFilesToStorage(dispatchedFileNames);
+        uploadFilesToStorage(rejectedFileNames);
+
+        // ensure files have been uploaded
+        assertThat(containerClient.listBlobs().stream().map(BlobItem::getName))
+            .containsExactlyInAnyOrder((String[]) addAll(dispatchedFileNames, rejectedFileNames));
+
+        createEnvelopes(DISPATCHED, dispatchedFileNames);
+        createEnvelopes(REJECTED, rejectedFileNames);
+
+        // when
+        containerCleaner.process(CONTAINER_NAME);
+
+        // then
+        assertThat(containerClient.listBlobs().stream().map(BlobItem::getName))
+            .containsExactlyInAnyOrder(rejectedFileNames);
+        assertFilesIsDeleteState(true, dispatchedFileNames);
+        assertFilesIsDeleteState(false, rejectedFileNames);
+    }
+
+    @Test
+    public void should_handle_non_existing_file() {
+        // given
+        // 2 envelopes in the database
+        String[] dispatchedFileNames = new String[]{TEST_1, TEST_3};
+        createEnvelopes(DISPATCHED, dispatchedFileNames);
+
+        // but only 1 file uploaded
+        uploadFilesToStorage(TEST_1);
+
+        // ensure file have been uploaded
+        assertThat(containerClient.listBlobs()).hasSize(1);
+        assertThat(containerClient.listBlobs().stream().map(BlobItem::getName))
+            .containsExactlyInAnyOrder(TEST_1);
+
+        // when
+        containerCleaner.process(CONTAINER_NAME);
+
+        // then
+        assertThat(containerClient.listBlobs()).isEmpty();
+        assertFilesIsDeleteState(true, dispatchedFileNames);
+    }
+
+    private void uploadFilesToStorage(String... fileNames) {
+        for (String fileName: fileNames) {
+            BlobClient blobClient = containerClient.getBlobClient(fileName);
+            blobClient.uploadFromFile("src/integrationTest/resources/storage/" + fileName);
+        }
+    }
+
+    private void createEnvelopes(Status status, String... fileNames) {
+        for (String fileName : fileNames) {
+            NewEnvelope envelope = new NewEnvelope(
+                CONTAINER_NAME,
+                fileName,
+                Instant.now(),
+                Instant.now(),
+                status
+            );
+            envelopeRepository.insert(envelope);
+        }
+    }
+
+    private void assertFilesIsDeleteState(boolean isDeleted, String... fileNames) {
```

Comment:
This method is never called with single file name as parameter.
It **could** be changed to take a list instead, and then we wouldn't have to deal with arrays everywhere.

Up to you.